test/extensions/firstorder/firstorder_settings.py

```diff
@@ -206,10 +210,38 @@
     ),
 ]
 
+###############################################################################
+#                         test setting: BatchNorm                             #
+###############################################################################
+FIRSTORDER_SETTINGS += [
+    {
+        "input_fn": lambda: rand(2, 3, 4),
+        "module_fn": lambda: initialize_training_false_recursive(
+            BatchNorm1d(num_features=3)
+        ),
+        "loss_function_fn": lambda: CrossEntropyLoss(reduction="mean"),
+        "target_fn": lambda: classification_targets((2, 4), 3),
+    },
+    {
+        "input_fn": lambda: rand(3, 2, 4, 3),
+        "module_fn": lambda: initialize_training_false_recursive(
+            BatchNorm2d(num_features=2)
+        ),
+        "loss_function_fn": lambda: CrossEntropyLoss(reduction="mean"),
+        "target_fn": lambda: classification_targets((3, 4, 3), 2),
+    },
+    {
+        "input_fn": lambda: rand(3, 3, 4, 1, 2),
+        "module_fn": lambda: initialize_training_false_recursive(
+            BatchNorm3d(num_features=3)
+        ),
+        "loss_function_fn": lambda: CrossEntropyLoss(reduction="mean"),
+        "target_fn": lambda: classification_targets((3, 4, 1, 2), 3),
+    },
+]
```

Comment:
Could you add a network multiple batch normalizations, e.g. `BN -> Linear -> ReLU -> BN`?

Reply:
added test

Reply:
Resolved